libraries/Microsoft.Bot.Builder.Azure/AzureBlobStorage.cs

```diff
@@ -15,8 +15,13 @@
 namespace Microsoft.Bot.Builder.Azure
 {
     /// <summary>
-    /// Models IStorage around a dictionary 
+    /// Models IStorage using Azure Storge Blobs
     /// </summary>
+    /// <remarks>
+    /// The AzureBlobStorage implements State's IStorage using a single Azure Storage Blob Container.
+    /// Each entity or StoreItem is serialized into a JSON string and stored in an individual text blob.
+    /// Each blob is named after the StoreItem key which is encoded and ensure it conforms a valid blob name.
+    /// </remarks>

```

Comment:
Thanks. This is makes thing much clearer. 

Can you add a sentence or two on how you'r using ETags and what the concurrency model is?